### PR TITLE
Support React Native 0.74

### LINKED ISSUE #221: Missing support for rn 0.74RC
Hey, while working on an [upgrade](https://github.com/Expensify/App/issues/37374) to react-native 0.74.0-rc.2 I have build errors related to use of deprecated Gradle features.

Confirmed on fresh react native app with rn-0.74.0-rc.2

`Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.`

```
/App/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java:14: error: cannot find symbol
import com.facebook.react.views.text.CustomLineHeightSpan;
                                    ^
  symbol:   class CustomLineHeightSpan
  location: package com.facebook.react.views.text
/App/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java:144: error: cannot find symbol
        CustomLineHeightSpan[] spans = ssb.getSpans(0, ssb.length(), CustomLineHeightSpan.class);
        ^
  symbol:   class CustomLineHeightSpan
  location: class MarkdownUtils
/App/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java:144: error: cannot find symbol
        CustomLineHeightSpan[] spans = ssb.getSpans(0, ssb.length(), CustomLineHeightSpan.class);
                                                                     ^
  symbol:   class CustomLineHeightSpan
  location: class MarkdownUtils
Note: /App/node_modules/@expensify/react-native-live-markdown/android/src/main/java/com/expensify/livemarkdown/MarkdownFontFamilySpan.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':expensify_react-native-live-markdown:compileDebugJavaWithJavac'.
```

### PR DESCRIPTION
**Pull request by tomekzaw**
### Details

Closes #221.

### Related Issues

GH_LINK

### CODE REVIEW

**Comment by tomekzaw** [comment]
Do not merge this until E/App upgrades to RN 0.74.

**Comment by roryabraham** [comment]
Asked about this in slack: https://expensify.slack.com/archives/C06BDSWLDPB/p1720805115823789

**Comment by cpetzel** [comment]
Is there an estimate for when this can get merged?

**Comment by tomekzaw** [comment]
We will merge this PR once E/App gets upgraded to 0.74.

**Comment by MrRefactor** [comment]
@tomekzaw ready for merge, E/App upgraded to 0.75

**Comment by roryabraham** [comment]
conflicts

**Comment by tomekzaw** [comment]
@roryabraham Conflicts resolved

File: src/MarkdownTextInput.web.tsx
Comment (tomekzaw) [inline src/MarkdownTextInput.web.tsx:690]: Without this change we get the following TS error:
```
src/MarkdownTextInput.web.tsx:690:5 - error TS2322: Type '"default"' is not assignable to type 'CursorValue | undefined'.

690     cursor: 'default',
        ~~~~~~

  node_modules/react-native/Libraries/StyleSheet/StyleSheetTypes.d.ts:279:3
    279   cursor?: CursorValue | undefined;
          ~~~~~~
    The expected type comes from property 'cursor' which is declared here on type 'ViewStyle | TextStyle | ImageStyle'
```

**Review by j-piasecki — DISMISSED** [review]

**Review by j-piasecki — APPROVED** [review]

**Comment by pleerock** [comment]
@tomekzaw when we can expect a release with this change?

**Comment by tomekzaw** [comment]
@pleerock Each merged PR to live-markdown gets automatically released. This PR was released in 0.1.122.

There's also a PR that adds support for RN 0.75 (https://github.com/Expensify/react-native-live-markdown/pull/428) and I would love to merge it by tomorrow.

**Comment by pleerock** [comment]
thanks @tomekzaw !

**Pull request merged**
